sdks/python/apache_beam/io/source_test_utils.py

```diff
@@ -67,6 +72,40 @@ class ExpectedSplitOutcome(object):
   MUST_BE_CONSISTENT_IF_SUCCEEDS = 3
 
 
+class TestCaseWithTempDirCleanUp(unittest.TestCase):
```

Comment:
I'm not sure that a base class is the best abstraction to do this. Can this be a simple Python context manager type that creates a temp dir in enter and removes it in exit, and test cases can use it as a simple member variable? I also think it shouldn't be in source_test_utils.py because it's in no way specific to testing Source classes.

Reply:
@jkff I agree, making this a context manager would be better. Where would suggest putting it rather than source_test_utils?

Reply:
+1 for making it a context manager. apache_beam/testing/test_utils.py might be a good place to move it to.

Reply:
Sounds good, thanks!